clang/lib/3C/Utils.cpp

```diff
@@ -563,3 +563,10 @@ void getPrintfStringArgIndices(const CallExpr *CE, const FunctionDecl *Callee,
         Context.getTargetInfo(), false);
   }
 }
+
+int64_t getStmtIdWorkaround(const Stmt *St, const ASTContext &Context) {
+  // XXX If we wanted to produce IDs of smaller absolute value, we could
+  // probably fix the alignment of negative values returned by
+  // identifyKnownObject so we can successfully divide them by alignof(Stmt).
+  return Context.getAllocator().identifyKnownObject(St);
```

Comment:
I'm not sure what this is doing. The code for `getID`, which this call replaces, is 
```
 int64_t Stmt::getID(const ASTContext &Context) const {
   return Context.getAllocator().identifyKnownAlignedObject<Stmt>(this);
 }
```
What's the difference between `identifyKnownAlignedObject` and `identifyKnownObject`? The latter is more forgiving, somehow, and thus avoids an issue with overflow? The comment above this call is cryptic, to me.

Reply:
Most of this is explained in the Clang bug report, but I don't mind explaining it here too.  Expanded the comment.